CMakeLists.txt

```diff
@@ -266,24 +274,43 @@ if (viam_rust_utils_files)
     ${viam_rust_utils_file}
     ONLY_IF_DIFFERENT
   )
-elseif(NOT WIN32) # TODO(RSDK-10366): Currently, rust_utils is not published for windows, so don't even try downloading
+elseif(NOT WIN32 OR rust_utils_supported_windows_arch) # TODO(RSDK-10366): Currently, rust_utils is not published for windows aarch, so don't even try downloading
   set(lvru_system_name ${CMAKE_SYSTEM_NAME})
-  if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  if (CMAKE_SYSTEM_NAME STREQUAL "Cygwin")
+    set(lvru_system_name "windows")
+  elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     set(lvru_system_name "macosx")
   endif()
+
+  set(lvru_system_processor ${CMAKE_SYSTEM_PROCESSOR})
+  if (WIN32 AND rust_utils_supported_windows_arch)
+    message(WARNING "windows and rust-utils-supported architecture is true")
+    set(lvru_system_processor "x86_64")
+  endif()
+  message(WARNING "system processor is ${lvru_system_processor}")
+
+  set(shared_library_prefix ${CMAKE_SHARED_LIBRARY_PREFIX})
+  if (WIN32)
+    set(shared_library_prefix "lib")
+  endif()
+  message(WARNING "shared library prefix is ${CMAKE_SHARED_LIBRARY_PREFIX}")
+  message(WARNING "trying to download https://github.com/viamrobotics/rust-utils/releases/latest/download/libviam_rust_utils-${lvru_system_name}_${lvru_system_processor}${CMAKE_STATIC_LIBRARY_SUFFIX}")
```

Comment:
There is an inherent conflict in windows library naming between shared (DLL) and static, because if you have `foo.dll` then you need an "import library" for that DLL, and that is normally called `foo.lib`. But if you want a static library as well then you can't call it `foo.lib` because that would conflict with the DLL import library. So, often, the static library has a `foo_static.lib` to distinguish it from the dynamic one. Other conventions are possible, but there is no strict rule here. From what I understand, the C++ SDK prefers to obtain the static version. So, the suffix is going to depend on how rust-utils publishes the relevant artifacts.